en/ros2/user_guide.md

```diff
@@ -751,6 +751,115 @@ will generate topics under the namespaces:
+
+A service server is the entity that will accept a remote procedure request, perform some computation on it, and return the result.
+They simplify communication between ROS 2 nodes and PX4 by encapsulating and associating the request and response topics, and ensuring that replies are only returned to the specific requesting user.
+This is much easier that publishing the request, subscribing to the reply, and filtering out any unwanted responses.
+
+The service servers that are built into the PX4 [uxrce_dds_client](../modules/modules_system.md#uxrce-dds-client) module include:
+
+- `/fmu/vehicle_command` (definition: [`px4_msgs::srv::VehicleCommand`](https://github.com/PX4/px4_msgs/blob/main/srv/VehicleCommand.srv).)
+
+  This service can be called by ROS 2 applications to send PX4 [VehicleCommand](../msg_docs/VehicleCommand.md) uORB messages and receive PX4 [VehicleCommandAck](../msg_docs/VehicleCommandAck.md) uORB messages in response.
+
+Details and specific examples are provided in the following sections.
+
+
+### VehicleCommand service
+
+This can be used to send commands to the vehicle, such as "take off", "land", change mode, and "orbit", and receive a response.
+
+The service is available under `/fmu/vehicle_command`, which can be changed through custom namespaces just like the other PX4 topics.
+The service type is defined in [`px4_msgs::srv::VehicleCommand`](https://github.com/PX4/px4_msgs/blob/main/srv/VehicleCommand.srv) as:
+
+```txt
+VehicleCommand request
+---
+VehicleCommandAck reply
+```
+
+Users can make service requests by sending [VehicleCommand](../msg_docs/VehicleCommand.md) messages, and receive a [VehicleCommandAck](../msg_docs/VehicleCommandAck.md) message in response.
+The service ensures that only the `VehicleCommandAck` reply generated for the specific request made by the user is sent back.
+
+#### VehicleCommand Service Offboard Control Example
+
+A complete _offboard control_ example using the VehicleCommand service is provided by the [offboard_control_srv](https://github.com/PX4/px4_ros_com/blob/main/src/examples/offboard/offboard_control_srv.cpp) node available in the `px4_ros_com` package.
+
+The example closely follows the _offboard control_ example described in [ROS 2 Offboard Control Example](../ros2/offboard_control.md) but uses the `VehicleCommand` service to request mode changes, vehicle arming and vehicle disarming.
+
+Like for any other ROS 2 service client, the ROS 2 application needs to declare a service client of type `px4_msgs::srv::VehicleCommand`:
+
+```cpp
+rclcpp::Client<px4_msgs::srv::VehicleCommand>::SharedPtr vehicle_command_client_;
+```
+
+and then initialize it to the right ROS 2 service.
+As the application assumes the standard PX4 namespace is used, this results in
+
+```cpp
+vehicle_command_client_{this->create_client<px4_msgs::srv::VehicleCommand>("/fmu/vehicle_command")}
+```
+
+After that, the client can be used to send any vehicle command request.
```

Comment:
The code seems to also require that you first wait for the service:

```
while (!vehicle_command_client_->wait_for_service(1s)) {
			if (!rclcpp::ok()) {
```

Where does `wait_for_service()` come from?

Reply:
Shouldn't we say that? Note that I have no idea where this method is defined (?) I assume `wait_for_service()` is some kind of standard interface/API? Or is it our own interface that hides some other magic?

Assuming it is standard, you might do something like this:

```suggestion
::: info
It is good practice to make sure the that the server server is ready to accept requests before sending them. You can use `wait_for_service()` to check if the server is ready.
:::

After that, the client can be used to send any vehicle command request.
```